pkg/file/io.go

```diff
@@ -107,3 +107,18 @@ func SplitWriteAll(ctx context.Context, s Splitter, r io.Reader, l int64, toEncr
 	}
 	return addr, nil
 }
+
+type Loader interface {
+	// Load a reference in byte slice representation and return all content associated with the reference.
+	Load([]byte) ([]byte, error)
```

Comment:
Since this is in Bee project, I think that `Load`/`Save` functions should return `swarm.Address` directly.

And maybe allow passing context to both operations.

Reply:
this interface was made to be compatible with `manifest.Loader` in the manifest package. have a closer look that i am injecting this interface directly into the mantaray dependency on load and save. if you want it as an address then we need to change the interface in mantaray first

Reply:
I am fully aware this was the reason. My comment is that we should not force external interface into this project.

The interface in "manatary" project was created to be independent from some external project. That interface is how it is, and should not be changed.
I would rather we have another level of indirection instead of having something like this. Especially since some other parts of Bee project might want to use this package.